src/main/java/net/minecraftforge/fluids/BlockFluidBase.java

```diff
@@ -492,6 +499,25 @@ public IBlockState getExtendedState(@Nonnull IBlockState oldState, @Nonnull IBlo
                 }
             }
         }
+        
+        //check for downflow above corners
+        if(isCoveredWithFluid(worldIn, pos.add(-1, 0, -1)) || isCoveredWithFluid(worldIn, pos.add(-1, 0, 0)) || isCoveredWithFluid(worldIn, pos.add(0, 0, -1)))
+        {
+            corner[0][0] = 1;
+        }
+        if(isCoveredWithFluid(worldIn, pos.add(-1, 0, 0)) || isCoveredWithFluid(worldIn, pos.add(-1, 0, 1)) || isCoveredWithFluid(worldIn, pos.add(0, 0, 1)))
+        {
+            corner[0][1] = 1;
+        }
+        if(isCoveredWithFluid(worldIn, pos.add(0, 0, -1)) || isCoveredWithFluid(worldIn, pos.add(1, 0, -1)) || isCoveredWithFluid(worldIn, pos.add(1, 0, 0)))
+        {
+            corner[1][0] = 1;
+        }
+        if(isCoveredWithFluid(worldIn, pos.add(1, 0, 0)) || isCoveredWithFluid(worldIn, pos.add(0, 0, 1)) || isCoveredWithFluid(worldIn, pos.add(1, 0, 1)))
+        {
+            corner[1][1] = 1;
+        }
+        
```

Comment:
This looks a bit more expensive than necessary. I'd make one if-block per pos to check and do multiple assigns inside. The assigning is cheap, the check is much more expensive.

```
if (isCoveredWithFluid(worldIn, pos.add(-1, 0, 0))) {
   corner[0][0] = 1;
   corner[0][1] = 1;
}
if (corner[0][0] != 1 && isCoveredWithFluid(worldIn, pos.add(-1, 0, -1))) {
   corner[0][0] = 1;
...
```

or if you want better readablility:

```
{
    boolean n = isCoveredWithFluid...
    boolean ne = isCoveredWithFluid...
    boolean e = ...

    if (n || ne || e) {
        ...
    if (e || se || se ...
```